src/libraries/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -555,7 +556,19 @@ private static string GetXmlSerializerAssemblyName(Type type, string defaultName
 
         private static string GetTempAssemblyName(AssemblyName parent, string ns)
         {
-            return parent.Name + ".XmlSerializers" + (ns == null || ns.Length == 0 ? "" : "." + ns.GetHashCode());
+            return parent.Name + ".XmlSerializers" + (string.IsNullOrEmpty(ns) ? "" : $".{GetPersistentHashCode(ns)}");
+        }
+
+        private static uint GetPersistentHashCode(string value)
+        {
+            byte[] valueBytes = Encoding.UTF8.GetBytes(value);
+            byte[] hash = SHA512.Create().ComputeHash(valueBytes);
+            return ReadUInt32BigEndian(hash);
+        }
+
+        private static uint ReadUInt32BigEndian(byte[] value)
```

Comment:
I assume we include our own implementation of ReadUInt32BigEndian() here because the framework version is netstandard2.1, while this tool targets netstandard2.0? If that's the case, I think this is fine.

@HongGit, do we have any plans to update the targetframework for this tool, or are we sticking to netstandard2.0?

Reply:
For now, we are still sticking to netstandard2.0.  If there is a need to update, we could consider to.